scribejava-apis/src/main/java/com/github/scribejava/apis/slack/SlackJsonTokenExtractor.java

```diff
@@ -17,6 +23,37 @@ public static SlackJsonTokenExtractor instance() {
         return SlackJsonTokenExtractor.InstanceHolder.INSTANCE;
     }
 
+    @Override
+    public OAuth2AccessToken extract(Response response) throws IOException {
+        final String body = response.getBody();
+        Preconditions.checkEmptyString(body,
+                "Response body is incorrect. Can't extract a token from an empty string");
+
+        if (response.getCode() != 200) {
+            generateError(response);
+        }
+        return createToken(body);
+    }
+
+    // Overridden private method
+    private OAuth2AccessToken createToken(String rawResponse) throws IOException {
+
+        final JsonNode response = OBJECT_MAPPER.readTree(rawResponse);
+
+        final JsonNode expiresInNode = response.get("expires_in");
+        final JsonNode refreshToken = response.get(OAuthConstants.REFRESH_TOKEN);
+        final JsonNode scope = response.get(OAuthConstants.SCOPE);
+        final JsonNode tokenType = response.get("token_type");
+        final JsonNode globalAccessToken = response.get(OAuthConstants.ACCESS_TOKEN);
+
+        return createToken( //
+                globalAccessToken == null ? "" : globalAccessToken.asText(), // Avoid "access_token can't be null"
```

Comment:
Note - the underlying OAuth2AccessToken (for SlackOAuth2AccessToken->super()) has a Precondition for "access_token can't be null", so here it is provided with an empty string.
The tokenType will be null, so it is easy to deduce whether the string is meaningful.